moveit_core/constraint_samplers/include/moveit/constraint_samplers/default_constraint_samplers.h

```diff
@@ -69,6 +69,16 @@ class JointConstraintSampler : public ConstraintSampler
   JointConstraintSampler(const planning_scene::PlanningSceneConstPtr& scene, const std::string& group_name)
     : ConstraintSampler(scene, group_name)
   {
+    int rng_seed;
+    if (ros::param::get("~joint_constraint_sampler_random_seed", rng_seed))
+    {
+      ROS_DEBUG_STREAM_NAMED("constraint_samplers", "Creating random number generator with seed " << rng_seed);
+      random_number_generator_ = std::make_unique<random_numbers::RandomNumberGenerator>(rng_seed);
+    }
+    else
+    {
+      random_number_generator_ = std::make_unique<random_numbers::RandomNumberGenerator>();
+    }
```

Comment:
Please move the constructor into `default_constraint_samplers.cpp`. There's no need for exposing ros function calls in the headers.
Also you don't need to change the API, because you can just set the random seed parameter in the constructor member initialization (before `{}`).

Reply:
I saw you downvoted my explanation at https://github.com/ros-planning/moveit/pull/3112/files#r845836293. Please provide the full code for list initialization. I don't know of a way to sneak `ros::param::get()` if-else branch into the member initialization before {}

Reply:
I pushed a solution: just move the parameter lookup and creation of the RNG into a separate function to be called from the constructor's member initialization :wink:

Reply:
OK, that is a way indeed. I was focused on doing this purely in the member initialization. Thanks for it 👍🏼